rust/src/dcerpc/parser.rs

```diff
@@ -154,7 +154,7 @@ pub fn parse_dcerpc_bindack(i: &[u8]) -> IResult<&[u8], DCERPCBindAck> {
     let (i, _assoc_group) = take(4_usize)(i)?;
     let (i, sec_addr_len) = le_u16(i)?;
     let (i, _) = take(sec_addr_len)(i)?;
-    let (i, _) = cond((sec_addr_len + 2) % 4 != 0, |b| take(4 - (sec_addr_len + 2) % 4)(b))(i)?;
+    let (i, _) = cond((sec_addr_len.wrapping_add(2)) % 4 != 0, |b| take(4 - (sec_addr_len.wrapping_add(2)) % 4)(b))(i)?;
```

Comment:
i worry we might hide a real issue here. Is having `sec_addr_len` as 65534 or 65535 even valid? Should we instead have ` verify!` on its parsing?

Reply:
@inashivb do you know about this sec_addr field ?

Reply:
> i worry we might hide a real issue here. Is having `sec_addr_len` as 65534 or 65535 even valid?

Didn't even think about it while reviewing. But, just checked the docs and we have,
```C
 The IDL declaration of the bind_ack PDU is as follows:
    typedef struct {
        .
        .
        u_int32 assoc_group_id;     /* 20:04 returned assoc_group_id */
        port_any_t sec_addr;        /* 24:yy optional secondary address 
                                     * for process incarnation; local port
                                     * part of address only */
        .
    }
```
where, `port_any_t` is defined as

```C
typedef struct {
        u_int16 length;
        char   [size_is(length)] port_spec; /* port string spec */
        } port_any_t;
```

which is also in accordance with what we have in our code,

```rust
pub struct DCERPCBindAck {
    .                                            
    pub sec_addr_len: u16,
    .
}
```

so I'm thinking it is valid?

Reply:
How does it related the fields before? E.g. would it make sense if it is bigger than a length field or a max pdu size minus what we already consumed?

Reply:
> How does it related the fields before? E.g. would it make sense if it is bigger than a length field or a max pdu size minus what we already consumed?

OK. I was only thinking in terms of int limits. I now understood what you meant. Indeed, it won't make sense and we should verify if it is within max PDU size limits..

Reply:
Some remarks without knowing everything about DCERPC :
- this is over TCP (so no 65k limit from UDP)
- Fuzzer managed to reach this, meaning it had a 65k payload. If the PDU was smaller, `take(sec_addr_len)(i)?;` should have failed and there is a bigger bug like https://redmine.openinfosecfoundation.org/issues/5018